### ISSUE TITLE
WQ: Add API for Python Function Evaluation

### ISSUE THREAD

**Issue by dthain** [cooperative-computing-lab/cctools#2600]
Good starter project for new student:

A common recent use case for WQ is to execute a single python function already defined in the main program.  This is quite doable but requires a fair amount of boilerplate to dill the function, dill the arguments, run a remote python script, bring the output back, undill the output and so forth.  See [wq_ensemble](https://github.com/cooperative-computing-lab/work-queue-examples/blob/master/wq_ensemble/ensemble.py) for an example.

Fix this up by introducing a new task type in the WQ Python API that encompasses the whole procedure for the user.

So, if the user ordinarily runs a function with arguments like this:
```
result = func( arg )
```

Then, that function should be executable via WQ like this:
```
task = wq.PyTask( func, arg )  # Pass in function and argument as plain python values.
q.submit(task);  # Submit and run task as normal.
task = q.wait();
result = task.pyresult;    # Gives back a proper python variable, not a string.
```

Of course, this works in the same way behind the scenes:

**Comment by dthain** [comment]
So, I didn't do my due diligence in Google.  :)  PyTask is already a well used name.  How about we just call this `work_queue.PythonTask` in parallel to `work_queue.Task` ?

**Comment by btovar** [comment]
I don't think it matters a lot, as it is an internal work_queue object.

**Comment by dthain** [comment]
PR #2610
